spec/factories/books.rb

```diff
@@ -0,0 +1,11 @@
+FactoryBot.define do
+  factory :book do
```

Comment:
This factory generates books without authors and categories. Add transient `authors_count { 1 }`, `categories_count { 1 }` and after_create hook for creating authors and categories.
Also, it's simpler to do books belongs_to one author and one category. So you can redo your migrations and models. It's yours to decide.